packages/discord.js/src/managers/GuildMemberManager.js

```diff
@@ -250,10 +255,9 @@ class GuildMemberManager extends CachedManager {
    * <info>The user must be a member of the guild</info>
    * @param {UserResolvable} user The member to edit
    * @param {GuildMemberEditData} data The data to edit the member with
-   * @param {string} [reason] Reason for editing this user
    * @returns {Promise<GuildMember>}
    */
-  async edit(user, data, reason) {
+  async edit(user, data) {
```

Comment:
calling edit with `{ nick, reason }` as data would use the wrong route ([again](https://github.com/discordjs/discord.js/pull/7151)), the check in [lines 287-289](https://github.com/discordjs/discord.js/pull/7448/files#diff-601b8954b3afea4b089cd0afd7b1c9679c874acaa1ea7c5a658319b5312e856aR287-R289) needs to be adapted